client/components/RecipeDetails/RecipeDetails.jsx

```diff
@@ -35,128 +34,113 @@ registerPlugin(
   FilePondPluginImageResize
 );
 
-class RecipeDetails extends Component {
-  static displayName = 'RecipeDetails';
-  static propTypes = {
-    className: PropTypes.string,
-    recipe: PropTypes.object,
-    recipeMods: PropTypes.arrayOf(PropTypes.object),
-    saveAlteration: PropTypes.func,
-    addPhoto: PropTypes.func,
-    photosLength: PropTypes.number,
-    client: PropTypes.instanceOf(ApolloClient)
+const RecipeDetails = ({
+  recipe,
+  addPhoto,
+  className,
+  photosLength,
+  recipeMods
+}) => {
+  const [errors, setErrors] = useState({});
+  const [edits, setEdit] = useState({});
+  const [editing, setEditing] = useState(!recipe ? true : false);
+  const [timeouts, setTimeoutValue] = useState({});
+
+  const containerRef = useRef(null);
+  const titleInputRef = useRef(null);
+  const descriptionInputRef = useRef(null);
+  const timeInputRef = useRef(null);
+  const skillInputRef = useRef(null);
+  const servingInputRef = useRef(null);
+  let pond = useRef(null);
+
+  const enableEditing = () => {
+    setEditing(true);
+    document.addEventListener('mousedown', handleClick);
   };
 
```

Comment:
Here i have refactored enableEdit to a single function and i have used if elses if that's ok with you or i can change this.

Reply:
sweet! I wonder if we could do away with checking if each ref's .current property exists and just use a switch case instead of if else statements

Reply:
I think you could also combine the enableEditing function with your new edit function. so if nothing is passed to enableEditing the default behavior simply turns on editing. But if a fieldname is passed into it, it enables editing and focuses the input.

Reply:
Got it, working on it.

Reply:
I have combined edit and enableEdit functions but would using switch make much difference here ?

Reply:
They'd be functionally equivalent; just a suggestion for readability and consistency (since switch cases are used elsewhere in the component). I also think checking the .current property might be an unnecessary carryover from the old code.

Reply:
Got it.